src/main/java/tc/oc/pgm/gui/InventoryGUI.java

```diff
@@ -0,0 +1,132 @@
+package tc.oc.pgm.gui;
+
+import com.google.common.collect.Sets;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.bukkit.Bukkit;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+import tc.oc.component.render.ComponentRenderers;
+import tc.oc.component.types.PersonalizedTranslatable;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.util.StringUtils;
+
+public abstract class InventoryGUI {
+
+  /*
+   * Code extracted from PickerMatchModule to allow for reuse
+   */
+
+  public Set<MatchPlayer> viewing = Sets.newHashSet();
+
+  public String title; // Title of the inventory
```

Comment:
Keep it private or protected, please.

Reply:
Thanks for the suggestions @TheMolkaPL !
I have to head out for a few hours, but I will make your suggested changes as soon as I’m back. 😄